tests/manage/pv_services/test_rbd_pv_encryption.py

```diff
@@ -0,0 +1,213 @@
+import logging
+import pytest
+
+from ocs_ci.framework.testlib import (
+    ManageTest,
+    tier1,
+    skipif_ocs_version,
+)
+from ocs_ci.helpers.helpers import (
+    create_unique_resource_name,
+    create_pods,
+)
+from ocs_ci.ocs import constants, defaults
+from ocs_ci.ocs.exceptions import (
+    CommandFailed,
+    KMSResourceCleaneupError,
+    ResourceNotFoundError,
+)
+from ocs_ci.utility import kms
+from ocs_ci.ocs.ocp import OCP
+
+log = logging.getLogger(__name__)
+
+
+@pytest.mark.parametrize(
+    argnames=["kv_version"],
+    argvalues=[
+        pytest.param("v1", marks=pytest.mark.polarion_id("OCS-2585")),
+        pytest.param("v2", marks=pytest.mark.polarion_id("OCS-2592")),
+    ],
+)
+@skipif_ocs_version("<4.7")
+class TestRbdPvEncryption(ManageTest):
+    """
+    Test to verify RBD PV encryption
+
+    """
+
+    @pytest.fixture(autouse=True)
+    def setup(self, kv_version):
+        """
+        Setup csi-kms-connection-details configmap
+
+        """
+        # Initialize Vault
+        self.vault = kms.Vault()
+        self.vault.gather_init_vault_conf()
+        self.vault.update_vault_env_vars()
+
+        # Check if cert secrets already exist, if not create cert resources
+        ocp_obj = OCP(kind="secret", namespace=constants.OPENSHIFT_STORAGE_NAMESPACE)
+        try:
+            ocp_obj.get_resource(resource_name="ocs-kms-ca-secret", column="NAME")
+        except CommandFailed:
```

Comment:
what if the command fails due to another reason? Maybe examine the exception and look for a certain string that says that the resources does not exist

Reply:
Changed the code to check for "not found" in the error message